src/rust/madara/template/src/template.rs

```diff
@@ -0,0 +1,344 @@
+		let obj = item.as_node();
+
+		let id = obj
+			.select("a")
+			.attr("href")
+			.read()
+			.replace(&data.base_url.clone(), "")
+			.replace(&data.source_path.clone(), "")
+			.replace("/", "");
+		let title = obj.select("a").attr("title").read();
+		let cover = get_image_url(obj.select("img"));
+
+		let genres = obj.select("div.post-content_item div.summary-content a");
+		if genres.text().read().to_lowercase().contains("novel") {
+			continue;
+		}
+
+		manga.push(Manga {
+			id,
+			cover,
+			title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
+			url: String::new(),
+			categories: Vec::new(),
+			status: MangaStatus::Unknown,
+			nsfw: MangaContentRating::Safe,
+			viewer: MangaViewer::Default,
+		});
+		has_more = true;
+	}
+
+	Ok(MangaPageResult { manga, has_more })
+}
+
+pub fn get_series_page(data: MadaraSiteData, listing: &str, page: i32) -> Result<MangaPageResult> {
+	let url = data.base_url.clone() + "/wp-admin/admin-ajax.php";
+
+	let body_content =  format!("action=madara_load_more&page={}&template=madara-core%2Fcontent%2Fcontent-archive&vars%5Bpaged%5D=1&vars%5Borderby%5D=meta_value_num&vars%5Btemplate%5D=archive&vars%5Bsidebar%5D=full&vars%5Bpost_type%5D=wp-manga&vars%5Bpost_status%5D=publish&vars%5Bmeta_key%5D={}&vars%5Border%5D=desc&vars%5Bmeta_query%5D%5Brelation%5D=OR&vars%5Bmanga_archives_item_layout%5D=big_thumbnail", &page-1, listing);
+
+	let req = Request::new(url.clone().as_str(), HttpMethod::Post)
+		.body(body_content.as_bytes())
+		.header("Content-Type", "application/x-www-form-urlencoded");
+
+	let html = req.html();
+
+	let mut manga: Vec<Manga> = Vec::new();
+	let mut has_more = false;
+	for item in html.select("div.page-item-detail").array() {
+		let obj = item.as_node();
+
+		let w_novel = obj.select(".web-novel").text().read();
+		if w_novel.len() > 0 {
+			continue;
+		}
+
+		let id = obj
+			.select("h3.h5 > a")
+			.attr("href")
+			.read()
+			.replace(&data.base_url.clone(), "")
+			.replace(&data.source_path.clone(), "")
+			.replace("/", "");
+
+		let title = obj.select("h3.h5 > a").text().read();
+		let img = get_image_url(obj.select("img"));
+
+		manga.push(Manga {
+			id,
+			cover: img,
+			title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
+			url: String::new(),
+			categories: Vec::new(),
+			status: MangaStatus::Unknown,
+			nsfw: MangaContentRating::Safe,
+			viewer: MangaViewer::Default,
+		});
+		has_more = true;
+	}
+
+	Ok(MangaPageResult { manga, has_more })
+}
+
+pub fn get_manga_listing(
+	data: MadaraSiteData,
+	_listing: Listing,
+	page: i32,
+) -> Result<MangaPageResult> {
+	if _listing.name == "Popular" {
+		return get_series_page(data, "_wp_manga_views", page);
+	}
+	if _listing.name == "Trending" {
+		return get_series_page(data, "_wp_manga_week_views_value", page);
+	}
+	return get_series_page(data, "_latest_update", page);
+}
```

Comment:
You're using this parameter, so it shouldn't be prefixed by an underscore.
```suggestion
pub fn get_manga_listing(
	data: MadaraSiteData,
	listing: Listing,
	page: i32,
) -> Result<MangaPageResult> {
	if listing.name == "Popular" {
		return get_series_page(data, "_wp_manga_views", page);
	}
	if listing.name == "Trending" {
		return get_series_page(data, "_wp_manga_week_views_value", page);
	}
	return get_series_page(data, "_latest_update", page);
}
```